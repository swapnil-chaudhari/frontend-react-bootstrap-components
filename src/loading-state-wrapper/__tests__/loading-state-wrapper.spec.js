import React from 'react';
import LoadingStateWrapper from '../loading-state-wrapper';
import renderShallow from 'render-shallow';


describe('<LoadingStateWrapper>', () => {
    describe('when isLoading is true', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(
                <LoadingStateWrapper isLoading={ true }>
                    <div>Hello World</div>
                </LoadingStateWrapper>
            ).output;
        });

        test('renders the <LoadingIndicator>', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when isLoading is false', () => {
        let component;
        beforeAll(() => {
            component = renderShallow(
                <LoadingStateWrapper isLoading={ false }>
                    <div>Hello World</div>
                </LoadingStateWrapper>
            ).output;
        });

        test('renders the child component', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
