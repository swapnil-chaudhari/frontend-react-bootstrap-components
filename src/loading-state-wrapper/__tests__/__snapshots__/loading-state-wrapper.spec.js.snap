// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingStateWrapper> when isLoading is false renders the child component 1`] = `
<div>
  Hello World
</div>
`;

exports[`<LoadingStateWrapper> when isLoading is true renders the <LoadingIndicator> 1`] = `<LoadingIndicator />`;
